Frontend completed

rohan9896 opened the pull request.

rohan9896 committed:
added videoCardExpand component

rohan9896 committed:
Feat: added AdCArd, CategoryScroller, NavBar, Sidebar, VideoCard, VideoCardExpand, VideoCardGrid, VideoSuggestionsCard, Context for liked/history/watchLater

rohan9896 committed:
feat: added local storage in notes

rohan9896 committed:
feat: added category context, landing page, routes, style: fixed card style

rohan9896 committed:
style: fixed margin of cards in mobile view

rohan9896 committed:
feat: added react router dependency

rohan9896 committed:
feat: added navlink

rohan9896 committed:
feat: added navigation to videos page when clicking on video card

rohan9896 committed:
feat: made videoCardGrid component reuseable by adding boolean props and fixed: warnings by adding key to every mapping

rohan9896 committed:
fixed: getting note from saved local storage

rohan9896 committed:
refractored some ;/indent/etc and style: changed styles of like and dislike btn

rohan9896 committed:
Update VideosData.js

rohan9896 committed:
feat: added clearHistoryBtn, liked, hostory, watch later page

rohan9896 committed:
fixed: video card resonsiveness and added views and monthsago from faker

rohan9896 committed:
added data

rohan9896 committed:
fixed: title overflowing and feat: added novideosfound component

rohan9896 committed:
feat: completed watchlater, liked and history

rohan9896 committed:
fix: changed add to watchlater reducer and feat: added toast

rohan9896 committed:
fixed: unique notes for different video id

rohan9896 committed:
changed background color and deleted unused app.css file

rohan9896 committed:
refractor some code

rohan9896 committed:
added suggested videos sidebar

rohan9896 committed:
chaged active font

rohan9896 committed:
Create _redirects

rohan9896 committed:
feat: added search functionality and chaged aspect ratio of video

rohan9896 committed:
added state variables in dependency array

rohan9896 committed:
changed title of website

sandeepmehta2155 commented on src/App.js:
```diff
@@ -1,11 +1,109 @@
-import './App.css';
+import { Routes, Route } from "react-router-dom";
+import CategoryScroller from "./components/CategoryScroller/CategoryScroller";
```
In app.js import files from index.js to make code look clean

rohan9896 replied:
Done

sandeepmehta2155 commented on src/App.js:
```diff
@@ -1,11 +1,109 @@
+            <>
+              {state.watchLaterArr.length === 0 ? (
+                <NoVideosFoundPage />
+              ) : (
```
Can use switch case but it's ok as logic is small

rohan9896 replied:
As there are only two things, i think ternary operator is fine

sandeepmehta2155 commented on src/components/CategoryScroller/CategoryScroller.js:
```diff
@@ -0,0 +1,94 @@
+  },
+  {
+    topic: "DataScience",
+    id: uuid(),
```
When data is reloaded, this might get change & effect your data updation

rohan9896 replied:
This id is only used for passing in the key prop of categories, it wouldn't make a diffrence

sandeepmehta2155 commented on src/components/NavBar/NavBar.js:
```diff
@@ -0,0 +1,56 @@
+              dispatch({type: "SEARCH_VIDEOS", payload: inputVal})
+              navigate(`/search?q=${inputVal}`);
+            }}>
+              <span role="img">🔍</span>
```
add a aria-labelled tag

rohan9896 replied:
done

sandeepmehta2155 reviewed (changes requested):
Amazing UI and very clean View

rohan9896 committed:
refractor: imported files from index.js

sandeepmehta2155 reviewed (approved):
Things had been reviewed, wishing all the best ahead

shubham-ghuge commented:
Hey Rohan,
The code and overall app looks good, I have reviewed the app here are some suggestions and fixes,
1. Can change/remove default favicon.
2. When i click directly on search button without adding any text the page goes blank.
3. Give fixed position to the header and category submenu.
4. when clicking on video the iframe size can be increased to maybe around 60vh or something.(Desktop view)
5. Search button style can be improved.

rohan9896 committed:
feat: added favicon, extracted reducer and actiontypes file, and fix: fixed page going blank on empty search

rohan9896 committed:
style: chnaged aspect ratio of video in mobile view and gave sticky position to category scroller

rohan9896 committed:
remove multiple uuids and added one directly in the jsx

rohan9896 committed:
style: changed aspect ratio property to height

rohan9896 commented:
> 1. Can change/remove default favicon.
 - done
> 2. When i click directly on search button without adding any text the page goes blank.
 - fixed
> 3. Give fixed position to the header and category submenu.
 - gave fixed position to category, fixed header wasn't looking good so avoided that
> 4. when clicking on video the iframe size can be increased to maybe around 60vh or something.(Desktop view)
 -  fixed
> 5. Search button style can be improved.
 - done

shubham-ghuge reviewed (approved):
Thank you for considering the changes Rohan,
I'm approving the PR.

rohan9896 committed:
feat: added backend

rohan9896 committed:
refactor: removed backend

Removed backend, will add that on separate branch so that i dont mess up my front end
